src/Catalyst.Core.Modules.Dfs/BlockExchange/Bitswap1.cs

```diff
@@ -0,0 +1,191 @@
+using Semver;
+using ProtoBufHelper = Lib.P2P.ProtoBufHelper;
+
+#pragma warning disable 0649 // disable warning about unassinged fields
+#pragma warning disable 0169// disable warning about unassinged fields
+
+namespace Catalyst.Core.Modules.Dfs.BlockExchange
+{
+    /// <summary>
+    ///   Bitswap Protocol version 1.0.0 
+    /// </summary>
+    public class Bitswap1 : IBitswapProtocol
+    {
+        private static ILog log = LogManager.GetLogger(typeof(Bitswap1));
+
+        /// <inheritdoc />
+        public string Name { get; } = "ipfs/bitswap";
+
+        /// <inheritdoc />
+        public SemVersion Version { get; } = new SemVersion(1, 0);
+
+        /// <inheritdoc />
+        public override string ToString() { return $"/{Name}/{Version}"; }
+
+        /// <summary>
+        ///   The <see cref="Bitswap"/> service.
+        /// </summary>
+        public Bitswap Bitswap { get; set; }
+
+        /// <inheritdoc />
+        public async Task ProcessMessageAsync(PeerConnection connection,
+            Stream stream,
+            CancellationToken cancel = default(CancellationToken))
+        {
+            var request = await ProtoBufHelper.ReadMessageAsync<Message>(stream, cancel).ConfigureAwait(false);
+
+            // There is a race condition between getting the remote identity and
+            // the remote sending the first wantlist.
+            await connection.IdentityEstablished.Task.ConfigureAwait(false);
+
+            log.Debug($"got message from {connection.RemotePeer}");
+
+            // Process want list
+            if (request.wantlist != null && request.wantlist.entries != null)
+            {
+                log.Debug("got want list");
+                foreach (var entry in request.wantlist.entries)
+                {
+                    var s = Base58.ToBase58(entry.block);
+                    Cid cid = s;
+                    if (entry.cancel)
+                    {
+                        // TODO: Unwant specific to remote peer
+                        Bitswap.Unwant(cid);
+                    }
+                    else
+                    {
+                        // TODO: Should we have a timeout?
+                        var _ = GetBlockAsync(cid, connection.RemotePeer, CancellationToken.None);
+                    }
+                }
+            }
+
+            // Forward sent blocks to the block service.  Eventually
+            // bitswap will here about and them and then continue
+            // any tasks (GetBlockAsync) waiting for the block.
+            if (request.blocks != null)
+            {
+                log.Debug("got some blocks");
+                foreach (var sentBlock in request.blocks)
+                    await Bitswap.OnBlockReceivedAsync(connection.RemotePeer, sentBlock);
+            }
+        }
+
+        private async Task GetBlockAsync(Cid cid, Peer remotePeer, CancellationToken cancel)
+        {
+            // TODO: Determine if we will fetch the block for the remote
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Complete the task associated to this 'TODO' comment.](https://app.codacy.com/app/catalyst-network/Catalyst.Framework/pullRequest?prid=4612924)